app/src/main/java/ch/epfl/sweng/SDP/utils/OnSwipeTouchListener.java

```diff
@@ -0,0 +1,59 @@
+package ch.epfl.sweng.SDP.utils;
+
+import android.content.Context;
+import android.view.GestureDetector;
+import android.view.GestureDetector.SimpleOnGestureListener;
+import android.view.MotionEvent;
+import android.view.View;
+
+/**
+ * Detects left and right swipes across a view.
+ */
+public class OnSwipeTouchListener implements View.OnTouchListener {
+
+    private final GestureDetector gestureDetector;
+
+    protected OnSwipeTouchListener(Context context) {
+        gestureDetector = new GestureDetector(context, new GestureListener());
+    }
+
+    public void onSwipeLeft() {
+        // The default swipe listener is not doing anything
```

Comment:
If they do nothing these methods could be either abstract or they could throw an UnsupportedOperationException

Reply:
No, otherwise the app would crash if the user swipes left. By default they do nothing and you override with the action you want.